scripts/sign-commits.sh

```diff
@@ -0,0 +1,35 @@
+#! /usr/bin/env bash
+# SPDX-FileCopyrightText: 2021 TQ Tezos <https://tqtezos.com/>
+#
+# SPDX-License-Identifier: LicenseRef-MIT-TQ
+
+# This script signs all commits in the current 'BUILDKITE_BRANCH'
+
+set -e
+
+git config user.name "serokell-bot" # necessary for pushing
+git config user.email "tezos-packaging@serokell.io"
+git fetch --all
+# We call this script only after a commit was pushed to the given branch, so it's safe to switch to it
+git switch "$BUILDKITE_BRANCH"
+git reset --hard origin/"$BUILDKITE_BRANCH"
+
+# Find if there are unsigned commits in the current branch
+# If there are grep matches then there are unsigned commits
+if git log --pretty="format:%G?" "origin/$BUILDKITE_PIPELINE_DEFAULT_BRANCH..$BUILDKITE_BRANCH" --first-parent | grep "N$"; then
+  echo "Found unsigned commits"
+  # Try to sign and push signed commits, retry in case of collision
+  while : ; do
+    git fetch --all
+    git reset --hard origin/"$BUILDKITE_BRANCH"
+    if ! git rebase --exec 'git commit --amend -n --gpg-sign="tezos-packaging@serokell.io" --no-edit' \
```

Comment:
This rebase basically adds a signature using the `tezos-packaging@serokell.io` key without editing the contents of the commit for each commit starting from `BUILDKITE_PIPELINE_DEFAULT_BRANCH` to `BUILDKITE_BRANCH` (all commits in the current branch)

Reply:
Oh yeah sure, sorry, I meant what might be the reasons the rebase fails, i.e. why you need to put it into an `if ! <rebase>` at all

Reply:
Ah, I've seen some weird `gnupg` behaviour, when signing failed without any visible reasons, e.g. see [this failure](https://buildkite.com/serokell/tezos-packaging/builds/1759#07c49d6a-03c4-40d2-a9f9-451e0c815c89/72-99), so I've added clean up just in case

Reply:
And also added `wait` to the pipeline configuration